src/bin/nydus-image/builder/mod.rs

```diff
@@ -71,10 +71,24 @@ fn dump_bootstrap(
     blob_writer: &mut ArtifactWriter,
 ) -> Result<()> {
     if let Some((_, blob_ctx)) = blob_mgr.get_current_blob() {
+        // Make sure blob id is updated according to blob hash if not specified by user.
         if blob_ctx.blob_id.is_empty() {
+            // `Blob::dump()` should have set `blob_ctx.blob_id` to referenced OCI tarball for
+            // ref-type conversion.
             assert!(!ctx.conversion_type.is_to_ref());
-            // Make sure blob id is updated according to blob hash if user not specified.
-            blob_ctx.blob_id = format!("{:x}", blob_ctx.blob_hash.clone().finalize());
+            if ctx.blob_inline_meta {
+                // Set special blob id for blob with inlined meta.
+                blob_ctx.blob_id = "x".repeat(64);
```

Comment:
It seems the blob id `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` will be persisted to bootstrap in the merge command.